crates/brainfuck_prover/src/brainfuck_air/mod.rs

```diff
@@ -90,26 +94,43 @@ pub fn lookup_sum_valid(
 ///
 /// Components are used by the prover as a `ComponentProver`,
 /// and by the verifier as a `Component`.
-pub struct BrainfuckComponents;
+pub struct BrainfuckComponents {
+    memory: MemoryComponent,
+}
 
 impl BrainfuckComponents {
     /// Initializes all the Brainfuck components from the claims generated from the trace.
     pub fn new(
-        _claim: &BrainfuckClaim,
-        _interaction_elements: &BrainfuckInteractionElements,
-        _interaction_claim: &BrainfuckInteractionClaim,
+        claim: &BrainfuckClaim,
+        interaction_elements: &BrainfuckInteractionElements,
+        interaction_claim: &BrainfuckInteractionClaim,
     ) -> Self {
-        todo!();
+        let memory_is_first_column = PreprocessedColumn::IsFirst(claim.memory.log_size);
+
+        let tree_span_provider =
+            &mut TraceLocationAllocator::new_with_preproccessed_columnds(&[memory_is_first_column]);
```

Comment:
Not related here but just PRed Stwo to fix the typo https://github.com/starkware-libs/stwo/pull/910